x/microtx/keeper/tokenized_account.go

```diff
@@ -0,0 +1,589 @@
+func (k Keeper) transferTokenizedAccountNFTFromModuleToAddress(ctx sdk.Context, contract common.Address, newOwner sdk.AccAddress) (*evmtypes.MsgEthereumTxResponse, error) {
+	// ABI: transferFrom(   address from,           address to,                uint256 tokenId)
+	var args = ToMethodArgs(types.ModuleEVMAddress, SDKToEVMAddress(newOwner), &AccountId)
+	return k.CallMethod(ctx, "transferFrom", types.TokenizedAccountNFT, types.ModuleEVMAddress, &contract, &big.Int{}, args...)
+}
+
+// addTokenizedAccountEntry Sets a new TokenizedAccount entry in the bech32 -> EVM NFT address mapping
+// accAddress - The account to Tokenize
+// nftAddress - The deployed TokenizedAccountNFT contract address
+func (k Keeper) addTokenizedAccountEntry(ctx sdk.Context, accAddress sdk.AccAddress, nftAddress common.Address) error {
+	store := ctx.KVStore(k.storeKey)
+	key := types.GetTokenizedAccountKey(accAddress)
+
+	if store.Has(key) {
+		return sdkerrors.Wrapf(types.ErrContractDeployment, "account %v already tokenized", accAddress.String())
+	}
+
+	store.Set(key, nftAddress.Bytes())
+	return nil
+}
+
+// queryTokenizedAccountOwner is used by the module to provide a convenient query interface, it calls the ERC721 ownerOf() function
+// with the only token used by TokenizedAccountNFTs (0x1) and returns the owner's eth address
+func (k Keeper) queryTokenizedAccountOwner(ctx sdk.Context, nftAddress common.Address) (*common.Address, error) {
+	// ABI: ownerOf(uint256 tokenId) public view virtual override returns (address)
+	var args = ToMethodArgs(&AccountId)
+	res, err := k.QueryEVM(ctx, "ownerOf", types.TokenizedAccountNFT, types.ModuleEVMAddress, &nftAddress, args...)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "unable to call ownerOf with arg0=1")
+	}
+	owner := common.BytesToAddress(res.Ret)
+	return &owner, nil
+}
+
+// queryTokenizedAccountThresholds is used by the module control tokenized account balances, it calls the
+// TokenizedAccountNFT getThresholds() function and formats the output into a useable type
+func (k Keeper) queryTokenizedAccountThresholds(ctx sdk.Context, nftAddress common.Address) ([]types.TokenizedAccountThreshold, error) {
+	// ABI: getThresholds() public virtual view returns (address[] memory, uint256[] memory)
+	res, err := k.QueryEVM(ctx, "getThresholds", types.TokenizedAccountNFT, types.ModuleEVMAddress, &nftAddress)
+	if err != nil {
+		return nil, sdkerrors.Wrap(err, "unable to call getThresholds with no arguments")
+	}
+
+	// The thresholds are returned as 2 dynamic arrays of 32 byte values. The contract should ensure the array lengths are the same
+	// however that is checked here.
+	// Dynamic values have a head value = offset where their info + elements begin. The info gives the number of elements in the array.
+	// So we want to see [ addresses off. | amounts off | num addresses | addr 1 | ... | addr n | num amounts | amt 1 | ... | amt n ]
+	thresholds := []types.TokenizedAccountThreshold{}
+	thresholdBytes := res.Ret
```

Comment:
you should have access to the geth abi decoder here? probably best to use it to do the decoding.

Reply:
Fixed